src/ArrayOfArrays.hpp

```diff
@@ -154,64 +108,79 @@ class ArrayOfArrays : protected ArrayOfArraysView< T, INDEX_TYPE >
 
-    // Reinterpret cast to ArrayOfArraysView so that we don't have to include ArrayOfSets.hpp.
-    ParentClass && srcView = reinterpret_cast< ParentClass && >( src );
+  /**
+   * @brief @return Return a reference to *this converted to ArrayOfArraysView<T, INDEX_TYPE const, true>.
+   * @note Duplicated for SFINAE needs.
+   */
+  LVARRAY_HOST_DEVICE constexpr inline
+  ArrayOfArraysView< T, INDEX_TYPE const, true > const & toViewSemiConst() const LVARRAY_RESTRICT_THIS
+  { return ParentClass::toViewSemiConst(); }
 
-    m_numArrays = srcView.m_numArrays;
-    srcView.m_numArrays = 0;
+  /**
+   * @brief @return Return a reference to *this converted to ArrayOfArraysView<T const, INDEX_TYPE const, true>.
+   * @note Duplicated for SFINAE needs.
+   */
+  LVARRAY_HOST_DEVICE constexpr inline
+  ArrayOfArraysView< T const, INDEX_TYPE const, true > const & toViewConst() const LVARRAY_RESTRICT_THIS
+  { return ParentClass::toViewConst(); }
 
-    m_offsets = std::move( srcView.m_offsets );
-    m_sizes = std::move( srcView.m_sizes );
-    m_values = std::move( srcView.m_values );
-  }
+  /**
+   * @brief @return Return the number of arrays.
+   * @note Duplicated for SFINAE needs.
+   */
+  inline
+  INDEX_TYPE size() const LVARRAY_RESTRICT_THIS
+  { return ParentClass::size(); }
 
   /**
    * @brief Reserve space for the given number of arrays.
-   * @param [in] newCapacity the new minimum capacity for the number of arrays.
+   * @param newCapacity the new minimum capacity for the number of arrays.
    */
   void reserve( INDEX_TYPE const newCapacity )
   { ParentClass::reserve( newCapacity ); }
 
   /**
    * @brief Reserve space for the given number of values.
-   * @param [in] newValueCapacity the new minimum capacity for the number of values across all arrays.
+   * @param newValueCapacity the new minimum capacity for the number of values across all arrays.
    */
   void reserveValues( INDEX_TYPE const newValueCapacity )
   { ParentClass::reserveValues( newValueCapacity ); }
 
   /**
    * @brief Set the number of arrays.
-   * @param [in] newSize the new number of arrays.
-   * @note We need this method in addition to the following resize method because of SFINAE requirements.
-   */
-  void resize( INDEX_TYPE const numArrays )
-  { ParentClass::resize( numArrays, 0 ); }
-
-  /**
-   * @brief Set the number of arrays.
-   * @param [in] newSize the new number of arrays.
-   * @param [in] defaultArrayCapacity the default capacity for each new array.
+   * @param newSize the new number of arrays.
+   * @param defaultArrayCapacity the default capacity for each new array.
    */
-  void resize( INDEX_TYPE const numArrays, INDEX_TYPE const defaultArrayCapacity )
-  { ParentClass::resize( numArrays, defaultArrayCapacity ); }
+  void resize( INDEX_TYPE const newSize, INDEX_TYPE const defaultArrayCapacity=0 )
+  { ParentClass::resize( newSize, defaultArrayCapacity ); }
```

Comment:
Does name newSize seems a little misleading? Not that I liked numArrays either...but since we are resizing the first dimension of the ArrayOfArrays (which is a sort of 2d array), I would suggest
```suggestion
  void resize( INDEX_TYPE const numSubArrays, INDEX_TYPE const defaultArrayCapacity=0 )
  { ParentClass::resize( numSubArrays, defaultArrayCapacity ); }
```